listbox_key_command is using the wrong path length constant

> `MAX_PATH` is undefined in Linux and builds fail with
> 
> 
> 
> ```
> 
> dxx-redux/d1/main/newmenu.c:1854:34: error: ‘MAX_PATH’ undeclared (first use in this function)
> 
>  1854 |         static char ascii_buffer[MAX_PATH];
> 
>       |                                  ^~~~~~~~
> 
> ``` 

 _Originally posted by @adolfogp in [b6e7208](https://github.com/dxx-redux/dxx-redux/commit/b6e7208c2978f1858266751f470fcb4f8d369f82#r151230448)_

This should be PATH_MAX, not MAX_PATH (which is a constant from one of the Windows headers).

--- a/d1/main/newmenu.c
+++ b/d1/main/newmenu.c
@@ -1851,7 +1851,7 @@ int listbox_mouse(window *wind, d_event *event, listbox *lb, int button)
 int listbox_key_command(window *wind, d_event *event, listbox *lb)
 {
 	// Note: statics are required to be zero-initialized in C, so we don't need to do it explicitly
-	static char ascii_buffer[MAX_PATH];
+	static char ascii_buffer[PATH_MAX];
 	static fix64 last_key_time;
 
 	// Reset the buffer after 3 seconds of inactivity

--- a/d2/main/newmenu.c
+++ b/d2/main/newmenu.c
@@ -1857,7 +1857,7 @@ int listbox_mouse(window *wind, d_event *event, listbox *lb, int button)
 int listbox_key_command(window *wind, d_event *event, listbox *lb)
 {
 	// Note: statics are required to be zero-initialized in C, so we don't need to do it explicitly
-	static char ascii_buffer[MAX_PATH];
+	static char ascii_buffer[PATH_MAX];
 	static fix64 last_key_time;
 
 	// Reset the buffer after 3 seconds of inactivity

